## Using Logestic Regression

Collect the data for HeartDisease data and use Logestic Regression to find Prediction and accuracy 

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [4]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression

Impoting the data using pandas library

In [6]:
df=pd.read_csv('Heart_Disease.csv')
df.head()

male  age  education  currentSmoker  cigsPerDay  BPMeds  prevalentStroke  \
0     1   39        4.0              0         0.0     0.0                0   
1     0   46        2.0              0         0.0     0.0                0   
2     1   48        1.0              1        20.0     0.0                0   
3     0   61        3.0              1        30.0     0.0                0   
4     0   46        3.0              1        23.0     0.0                0   

   prevalentHyp  diabetes  totChol  sysBP  diaBP    BMI  heartRate  glucose  \
0             0         0    195.0  106.0   70.0  26.97       80.0     77.0   
1             0         0    250.0  121.0   81.0  28.73       95.0     76.0   
2             0         0    245.0  127.5   80.0  25.34       75.0     70.0   
3             1         0    225.0  150.0   95.0  28.58       65.0    103.0   
4             0         0    285.0  130.0   84.0  23.10       85.0     85.0   

   TenYearCHD  
0           0  
1           0  
2           0  
3           1  
4           0

Finding the null values and dtypes 

In [7]:
df.isna().sum()

male                 0
age                  0
education          105
currentSmoker        0
cigsPerDay          29
BPMeds              53
prevalentStroke      0
prevalentHyp         0
diabetes             0
totChol             50
sysBP                0
diaBP                0
BMI                 19
heartRate            1
glucose            388
TenYearCHD           0
dtype: int64

Using the mean value to fill all the null value columns

In [8]:
df1=df.fillna(df.mean())

In [9]:
df1.isna().sum()

male               0
age                0
education          0
currentSmoker      0
cigsPerDay         0
BPMeds             0
prevalentStroke    0
prevalentHyp       0
diabetes           0
totChol            0
sysBP              0
diaBP              0
BMI                0
heartRate          0
glucose            0
TenYearCHD         0
dtype: int64

In [10]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4238 entries, 0 to 4237
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   male             4238 non-null   int64  
 1   age              4238 non-null   int64  
 2   education        4238 non-null   float64
 3   currentSmoker    4238 non-null   int64  
 4   cigsPerDay       4238 non-null   float64
 5   BPMeds           4238 non-null   float64
 6   prevalentStroke  4238 non-null   int64  
 7   prevalentHyp     4238 non-null   int64  
 8   diabetes         4238 non-null   int64  
 9   totChol          4238 non-null   float64
 10  sysBP            4238 non-null   float64
 11  diaBP            4238 non-null   float64
 12  BMI              4238 non-null   float64
 13  heartRate        4238 non-null   float64
 14  glucose          4238 non-null   float64
 15  TenYearCHD       4238 non-null   int64  
dtypes: float64(9), int64(7)
memory usage: 529.9 KB


All  the Data's are present in the numerical.

We split the data for train and test

In [19]:
x=df1.drop(['TenYearCHD'],axis=1)
y=df1['TenYearCHD']

## Scaling a Data 

In [27]:
#using Standard scaler method
from sklearn.preprocessing import StandardScaler
ss=StandardScaler()
X=ss.fit_transform(x)

In [28]:
x_train,x_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=10)

In [29]:
reg=LogisticRegression()
fit=reg.fit(x_train,y_train)

In [34]:
# predict 
y_pred=reg.predict(x_test)
pd.DataFrame(y_pred)

0
0    0
1    0
2    0
3    0
4    0
..  ..
843  0
844  0
845  0
846  0
847  0

[848 rows x 1 columns]

In [39]:
from sklearn.metrics import accuracy_score
acc=accuracy_score(y_test,y_pred)
acc

0.8691037735849056

we Got an 86% of Accuracy of our Model

## use confusion matrix to find performance

In [45]:
from sklearn.metrics import confusion_matrix
com=confusion_matrix(y_test,y_pred)
com

array([[731,   8],
       [103,   6]], dtype=int64)

our Value of the
    True Positive=731,
    False Positive=8,
    False Negative=103,
    True Negative=6.

##  Evaluate the model using other performance metrics

In [47]:
from sklearn.metrics import classification_report
print(classification_report(y_test,y_pred))

              precision    recall  f1-score   support

           0       0.88      0.99      0.93       739
           1       0.43      0.06      0.10       109

    accuracy                           0.87       848
   macro avg       0.65      0.52      0.51       848
weighted avg       0.82      0.87      0.82       848

